### DISCUSSION TITLE
[K+] Metron.Cloud partial integration

### DISCUSSION THREAD

**Discussion by Cducharme84** [Kareadita/Kavita#3165] [Ideas]
### Idea Description

With western comics there's lots of unique challenges with a full integration/usage of the data for all 3 major sources for English comics, CV has the API limit that would make it feel like Kavita's not really populating, GCD would require the downloading of the database by the admin and doing so on the regular, and Metron.Cloud is incredibly promising but is just new, so has a dearth of back catalog data making filling out your library with its data not yet feasible for those with non-major publisher or older catalogs in general.

That said, Metron.Cloud's API could prove very useful for Kavita+ in a limited manner, specifically with newer releases that they have data for. Comic Vine has many rules that prevent the data from being populated until after release of the issues from legal digital sources, this causes a lag where if you tag with data from then right after obtaining your comic you'll get a mostly blank xml with just the series/volume/issue filled in 99% of the time, sometimes the series even defaults to a very far off year (2099) which just looks bad, but this is a data quality issue not present with Metron as they allow entering all formally solicited issues.

Looking into their [API](https://metron.cloud/docs/) : I see they have an endpoint for publisher with "list" as an action. I propose every 24-48 hours Kavita+ interact with this endpoint to find the publishers present on Metron. These would then be recorded in the Kavita DB there are currently only 103 publishers on Metron so this should not be expensive even on first interaction. The endpoint has a "modified" attribute to filter by so updates can be even less expensive for Kavita/Metron itself.

From an admin standpoint, in the K+ sections for metadata there would be a username/password field (this is how the Metron API authenticates) for the user's DB account with Metron. And (optionally) a selector to select which year forward to monitor for publishers that are on Metron when added to the library.

These two changes/actions would be to facilitate when adding issues/series from a publisher on the list from Metron's database and with a release year greater than or equal to the selected year by the admin in the K+ screen, it will attempt to match the issue/series with Metron using the following in order of preference:

1. ComicVine ID: if there is a comicvine ID present for the issue or series and it matches the Metron database (which allows searching and matching with that off-site identifier) this can be considered a positive match.
2. Series name and volume start year: If this matches and Metron has an issue of the same number, consider a match.

If no matches found, whether at publisher (not present at all), by CV IDs when present (already recorded in kavita DB when present), or if series/issue is not present on Metron do nothing.

But if it is? Update the issue/series (only update Series metadata relevant to count etc after issue 1 but if this is issue 1 being added treat the newly grabbed data as if it were XML provided data, with optional lock to prevent scan from replacing with empty CV provided data) with the data from Metron.

This would be most useful for mylar users following current publishers as well as those adding digital purchases for new comic book day for publishers on the Metron list.

There are a few drawbacks:

This would require DB changes, which is always something to plan carefully and consider all options regarding.

Metron doesn't have a large data pool yet, so this is most useful for the largest American publishers, in fact they currently only allow American books. So this does limit utility for international audiences.

But I do think this is a great way to dip into the western comics potential of Kavita+. 

Also integration with Metron may see increased contributions to that data pool just through exposure as admins obtain their username and password, benefiting both K+ and the community in general.

### Idea Category

Feature Enhancement

### Duration of Using Kavita

1.5 or so years

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by majora2007** [comment]
I've been keeping my eye on Metron for a while now, as CV has very strict rates and a license that wouldn't allow integration. This is a very detailed write up and does a lot of leg work. I can't promise anything, but I'll add it to my Kavita+ backlog and give it a shot to see if it is feasible and a good UX.
